discovery/linode/mock_test.go

```diff
@@ -63,193 +61,41 @@ func (m *SDMock) HandleLinodeInstancesList() {
-      "label": "prometheus-linode-sd-exporter-3",
-      "group": "",
-      "status": "running",
-      "created": "2021-05-12T04:20:11",
-      "updated": "2021-05-12T04:20:11",
-      "type": "g6-standard-1",
-      "ipv4": [
-        "192.53.120.25"
-      ],
-      "ipv6": "2600:3c04::f03c:92ff:fe1a:fb68/128",
-      "image": "linode/ubuntu20.04",
-      "region": "ca-central",
-      "specs": {
-        "disk": 51200,
-        "memory": 2048,
-        "vcpus": 1,
-        "gpus": 0,
-        "transfer": 2000
-      },
-      "alerts": {
-        "cpu": 90,
-        "network_in": 10,
-        "network_out": 10,
-        "transfer_quota": 80,
-        "io": 10000
-      },
-      "backups": {
-        "enabled": false,
-        "schedule": {
-          "day": null,
-          "window": null
-        },
-        "last_successful": null
-      },
-      "hypervisor": "kvm",
-      "watchdog_enabled": true,
-      "tags": [
-        "monitoring"
-      ]
-    },
-    {
-      "id": 26837992,
-      "label": "prometheus-linode-sd-exporter-4",
-      "group": "",
-      "status": "running",
-      "created": "2021-05-12T04:22:06",
-      "updated": "2021-05-12T04:22:06",
-      "type": "g6-nanode-1",
-      "ipv4": [
-        "66.228.47.103",
-        "172.104.18.104",
-        "192.168.148.94"
-      ],
-      "ipv6": "2600:3c03::f03c:92ff:fe1a:fb4c/128",
-      "image": "linode/ubuntu20.04",
-      "region": "us-east",
-      "specs": {
-        "disk": 25600,
-        "memory": 1024,
-        "vcpus": 1,
-        "gpus": 0,
-        "transfer": 1000
-      },
-      "alerts": {
-        "cpu": 90,
-        "network_in": 10,
-        "network_out": 10,
-        "transfer_quota": 80,
-        "io": 10000
-      },
-      "backups": {
-        "enabled": false,
-        "schedule": {
-          "day": null,
-          "window": null
-        },
-        "last_successful": null
-      },
-      "hypervisor": "kvm",
-      "watchdog_enabled": true,
-      "tags": [
-        "monitoring"
-      ]
-    }
-  ],
-  "page": 1,
-  "pages": 1,
-  "results": 4
-}`,
-		)
+		switch fltr := r.Header.Get("X-Filter"); fltr {
+		case "{\"region\": \"us-east\"}":
+			fmt.Fprint(w, instancesUsEast)
+		case "{\"region\": \"ca-central\"}":
+			fmt.Fprint(w, instancesCaCentral)
+		default:
+			fmt.Fprint(w, instancesAll)
+		}
 	})
 }
 
-// HandleLinodeNeworkingIPs mocks linode networking ips endpoint.
-func (m *SDMock) HandleLinodeNeworkingIPs() {
+// HandleLinodeNetworking/ipv6/ranges mocks linode networking ipv6 ranges endpoint.
+func (m *SDMock) HandleLinodeNetworkingIPv6Ranges() {
+	m.Mux.HandleFunc("/v4/networking/ipv6/ranges", func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", tokenID) {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		w.Header().Set("content-type", "application/json; charset=utf-8")
+		w.WriteHeader(http.StatusOK)
+
+		switch fltr := r.Header.Get("X-Filter"); fltr {
```

Comment:
I'm leaving a single comment here, but this applies to each of these switch cases: 

This definitely improves the readability of the tests, but we're starting to depart pretty far from idiomatic Go testing:
https://go.dev/wiki/TableDrivenTests
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

I realize the existing test suite isn't table-driven or idiomatic either, but I think it would be better to try to bring this more in line with how Go recommends structuring tests vs departing further

Reply:
Is there any existing test in the prometheus codebase that uses TableDriveTests?

Reply:
converted to TableDrivenTests as per 145c7858d45b63b4a1ae53972240dca3afe154dc